src/overlays/actors/ovl_En_Knight/z_en_knight.c

```diff
@@ -205,237 +380,4015 @@ ActorInit En_Knight_InitVars = {
+                if (this->dmgEffectScale == 0.0f) {
+                    this->dmgEffectState = KNIGHT_DMGEFF_STATE_0;
+                    this->dmgEffectAlpha = 0.0f;
+                }
+            }
+            break;
+    }
+
+    this->jawRotationAmplitudeTarget = 0.0f;
+
+    if (this == sIgosInstance) {
+        if (sMinibossStartBgmCountdown != 0) {
+            sMinibossStartBgmCountdown--;
+            if (sMinibossStartBgmCountdown == 0) {
+                Audio_PlayBgm_StorePrevBgm(NA_BGM_MINI_BOSS);
+            }
+        }
 
-extern DamageTable D_809BDB04;
-extern DamageTable D_809BDB24;
-extern ColliderJntSphElementInit D_809BDB44[1];
-extern ColliderJntSphInit D_809BDB8C;
-extern ColliderJntSphElementInit D_809BDB68[1];
-extern ColliderJntSphInit D_809BDB9C;
-extern ColliderJntSphElementInit D_809BDBAC[1];
-extern ColliderJntSphInit D_809BDBD0;
-extern ColliderJntSphElementInit D_809BDBE0[2];
-extern ColliderJntSphInit D_809BDC28;
-extern ColliderCylinderInit D_809BDC38;
-extern ColliderCylinderInit D_809BDC64;
+        if (this->isHeadless) {
+            Math_ApproachF(&this->bodyAlpha, 60.0f, 1.0f, 8.0f);
+        } else {
+            Math_ApproachF(&this->bodyAlpha, 255.0f, 1.0f, 8.0f);
+        }
 
-extern UNK_TYPE D_060005A8;
-extern UNK_TYPE D_06001CDC;
-extern UNK_TYPE D_06002174;
-extern UNK_TYPE D_06003008;
-extern UNK_TYPE D_060031F0;
-extern UNK_TYPE D_06003650;
-extern UNK_TYPE D_060040E0;
-extern UNK_TYPE D_06004620;
-extern UNK_TYPE D_06004974;
-extern UNK_TYPE D_060079D4;
-extern UNK_TYPE D_06008524;
-extern UNK_TYPE D_06008D80;
-extern UNK_TYPE D_06009538;
-extern UNK_TYPE D_06009D8C;
-extern UNK_TYPE D_0600A530;
-extern UNK_TYPE D_0600AFAC;
-extern UNK_TYPE D_0600B5D4;
-extern UNK_TYPE D_0600BCF4;
-extern UNK_TYPE D_0600CDE0;
-extern UNK_TYPE D_0600DDCC;
-extern UNK_TYPE D_0600EA90;
-extern UNK_TYPE D_0600EF44;
-extern UNK_TYPE D_06010E98;
-extern UNK_TYPE D_06011298;
-extern UNK_TYPE D_06013020;
-extern UNK_TYPE D_06018BC4;
-extern UNK_TYPE D_06020950;
-extern UNK_TYPE D_06022728;
-extern UNK_TYPE D_06022CAC;
+        EnKnight_UpdateEffects(this, play);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Knight/func_809B20F0.s")
+        if (this->roomLightingState != 0) {
+            u16 time;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Knight/func_809B21F4.s")
+            play->envCtx.lightSettingOverride = 0;
+            play->envCtx.lightBlendOverride = 2;
+            if (this->roomLightingState == 1) {
+                play->envCtx.lightSetting = 8;
+                play->envCtx.prevLightSetting = 12;
+            } else {
+                play->envCtx.lightSetting = 0;
+                play->envCtx.prevLightSetting = 10;
+            }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Knight/func_809B22CC.s")
+            time = gSaveContext.save.time;
+            if ((s32)time > 0x8000) {
+                time = (0xFFFF - time) & 0xFFFF;
```

Comment:
```suggestion
                time = ((DAY_LENGTH - 1) - time) & 0xFFFF;
```
Does this work like in boss_06? Also is the mask necessary, or possibly an explicit `u16` cast also works (though not sure if that is better then the mask)?